testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeIdentityColumnCompatibility.java

```diff
@@ -0,0 +1,324 @@
+
+            assertThat((String) onDelta().executeQuery("SHOW CREATE TABLE default." + tableName).getOnlyValue())
+                    .contains("b BIGINT GENERATED ALWAYS AS IDENTITY");
+            onDelta().executeQuery("INSERT INTO default." + tableName + " (a, c) VALUES (0, 2)");
+
+            QueryAssert.Row expected = row(0, 1, 2);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName)).containsOnly(expected);
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName)).containsOnly(expected);
+        }
+        finally {
+            dropDeltaTableWithRetry("default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, DELTA_LAKE_EXCLUDE_113, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testIdentityColumnFeature()
+    {
+        String tableName = "test_identity_column_feature_" + randomNameSuffix();
+
+        onDelta().executeQuery("CREATE TABLE default." + tableName +
+                "(data INT, col_identity BIGINT GENERATED ALWAYS AS IDENTITY)" +
+                "USING DELTA " +
+                "LOCATION 's3://" + bucketName + "/" + "databricks-compatibility-test-" + tableName + "'" +
+                "TBLPROPERTIES ('delta.feature.identityColumns'='supported')");
+        try {
+            assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, 1)"))
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("UPDATE delta.default." + tableName + " SET data = 1"))
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + tableName))
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO delta.default." + tableName + " t USING delta.default." + tableName + " s " +
+                    "ON (t.data = s.data) WHEN MATCHED THEN UPDATE SET data = 1"))
+                    .hasMessageMatching(".* Table .* requires Delta Lake writer version 7 which is not supported");
+        }
+        finally {
+            dropDeltaTableWithRetry("default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testWritesToTableWithIdentityColumnFails()
```

Comment:
What is the difference between this test and the test above?

Reply:
In the above test, https://github.com/delta-io/delta/blob/master/PROTOCOL.md#valid-feature-names-in-table-features the protocol version points the min writer version to be `7`

```
{
  "protocol": {
    "minReaderVersion": 1,
    "minWriterVersion": 7,
```

while in this test, the inferred writer version will be `6`.